src/main/java/org/aksw/iguana/commons/io/ByteArrayListInputStream.java

```diff
@@ -95,6 +99,18 @@ public int readNBytes(byte[] b, int off, int len) throws IOException {
         return read(b, off, len, 0);
```

Comment:
Throw IOExceltion if `length>MAX_ARRAY_SIZE. Also in read(byte[] b, int off, int len)`.
From how I understand https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html the length must only be truncated if the end is reached.